src/common.js

```diff
@@ -274,7 +283,24 @@ function wrap(cmd, fn, options) {
         });
         if (!config.noglob && options && typeof options.idx === 'number')
           args = args.slice(0, options.idx).concat(expand(args.slice(options.idx)));
-        retValue = fn.apply(this, args);
+        if (config.async || callback) {
+          process.nextTick(function() {
+            var r = fn.apply(this, args);
```

Comment:
@ariporad I just tried this example. I saw:

> It's been 1 second!
> It's been 10 seconds!

Also, `exec()`'s async version is totally different than the rest of the commands.

Another example:

```
> setTimeout(() => console.log('It\'s been 1 second!'), 5000);
> ls({async: true}, function() { echo('hi'); });
```

Output (as I would expect, since `ls()` is super fast) :

> hi
> It's been 1 second!